pkg/checks/base.go

```diff
@@ -56,6 +56,9 @@ type Check interface {
 	Schema() (*openapi3.SchemaRef, error)
 	// GetMetricCollectors allows the check to provide prometheus metric collectors
 	GetMetricCollectors() []prometheus.Collector
+	// RemoveLabelledMetrics allows the check to remove the prometheus metrics
+	// of the check whose `target` label matches the passed value
+	RemoveLabelledMetrics(target string) error
 }
```

Comment:
I think we should split this interface up into multiple interfaces that can be implemented by a check in the near future. 

For example if a check cannot expose any prometheus metrics, it should not need to implement these methods. I think the best way forward would be to move everything out of the `Check` interface and then composing the check by using the new smaller interfaces:
```go
// Runner interface for running and shutting down the check
type Runner interface {
	Run(ctx context.Context, cResult chan ResultDTO) error
	Shutdown()
}

// Configurable interface for setting and getting configuration
type Configurable interface {
	UpdateConfig(config Runtime) error
	GetConfig() Runtime
}

// Identifiable interface for getting the name of the check
type Identifiable interface {
	Name() string
}

// SchemaProvider interface for providing schema
type SchemaProvider interface {
	Schema() (*openapi3.SchemaRef, error)
}

// MetricProvider interface for providing metric collectors
type MetricProvider interface {
	GetMetricCollectors() []prometheus.Collector
	RemoveLabelledMetrics(target string) error
}

// Check implementations are expected to perform specific monitoring tasks and report results.
type Check interface {
	Runner
	Configurable
	Identifiable
	SchemaProvider
	MetricProvider
}
```

This way we can have a more flexible interface that can be implemented by checks that do not need to implement all the methods.

Reply:
That looks like a good idea, at first glance. I'll take a swing at it and see where it goes. But not as part of this PR, as this would totally be outta scope. 😉